macos/ReactTestApp/AppDelegate.swift

```diff
@@ -237,18 +238,19 @@ extension AppDelegate {
             let modalFrame = NSRect(size: WindowSize.modalSize)
             rootView.frame = modalFrame
 
-            var token: NSObjectProtocol?
-            token = NotificationCenter.default.addObserver(
+            contentDidAppearToken = NotificationCenter.default.addObserver(
                 forName: .RCTContentDidAppear,
                 object: rootView,
                 queue: nil,
-                using: { _ in
+                using: { [weak self] _ in
                     #if USE_FABRIC
                     rootView.frame = modalFrame
                     #else
                     (rootView as? RCTRootView)?.contentView.frame = modalFrame
                     #endif
-                    NotificationCenter.default.removeObserver(token!)
+                    if let token = self?.contentDidAppearToken {
+                        NotificationCenter.default.removeObserver(token)
```

Comment:
Do you need to call removeObservor at all? I thought those were auto removed on de-init?

Reply:
https://developer.apple.com/documentation/foundation/notificationcenter/1413994-removeobserver

> If your app targets iOS 9.0 and later or macOS 10.11 and later, and you used [addObserver(_:selector:name:object:)](https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver), you do not need to unregister the observer. If you forget or are unable to remove the observer, the system cleans up the next time it would have posted to it.

Reply:
We're using [addObserver(forName:object:queue:using:)](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver) though:

> If you used [addObserver(forName:object:queue:using:)](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver) to create your observer, you should call this method or [removeObserver(_:name:object:)](https://developer.apple.com/documentation/foundation/notificationcenter/1407263-removeobserver) before the system deallocates any object that [addObserver(forName:object:queue:using:)](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver) specifies.

Or are you saying we should use [addObserver(_:selector:name:object:)](https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver) instead? I think we can do that.